wa/workloads/motionmark/__init__.py

```diff
@@ -0,0 +1,77 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import re
+
+from wa import UiautoWorkload, Parameter
+from wa.framework.exception import ValidationError, WorkloadError
+from wa.utils.types import list_of_strs
+from wa.utils.misc import unique
+
+
+class Motionmark(UiautoWorkload):
+
+    name = 'motionmark'
+    regex_matches = [re.compile(r'Multiply Score (.+)'),
+                     re.compile(r'Canvas Score (.+)'),
+                     re.compile(r'Leaves Score (.+)'),
+                     re.compile(r'Paths Score (.+)'),
+                     re.compile(r'Canvas Lines Score (.+)'),
+                     re.compile(r'Focus Score (.+)'),
+                     re.compile(r'Images Score (.+)'),
+                     re.compile(r'Design Score (.+)'),
+                     re.compile(r'Suits Score (.+)')]
+    score_regex = re.compile(r'.*?([\d.]+).*')
+    description = '''
+    A workload to execute the motionmark web based benchmark
+
+    Test description:
+    1. Open browser application
+    2. Navigate to the motionmark website - http://browserbench.org/motionmark/
+    3. Execute the benchmark
+
+    '''
+
+    def __init__(self, target, **kwargs):
+        super(Motionmark, self).__init__(target, **kwargs)
+        self.gui.timeout = 1500
+
+    def setup(self, context):
+        super(Motionmark, self).setup(context)
+        url = 'am start -a android.intent.action.VIEW -d https://browserbench.org/MotionMark/'
+        self.target.execute(url)
+
+    def update_output(self, context):
+        super(Motionmark, self).update_output(context)
+        expected_results = len(self.regex_matches)
+        logcat_file = context.get_artifact_path('logcat')
+        with open(logcat_file) as fh:
+            for line in fh:
+                for regex in self.regex_matches:
+                    match = regex.search(line)
+                    # Check if we have matched the score string in logcat
+                    if match:
+                        score_match = self.score_regex.search(match.group(1))
+                        # Check if there is valid number found for the score.
+                        if score_match:
+                            result = float(score_match.group(1))
+                        else:
+                            result = 'NaN'
+                        entry = regex.pattern.rsplit(None, 1)[0]
+                        context.add_metric(entry, result, 'Score', lower_is_better=False)
+                        expected_results -= 1
+        if expected_results > 0:
```

Comment:
Variable name does not match its function. Rename to something like `num_unprocessed_results`. Or even better, why not just throw an exception if `match` above is `None` and remove the need for this variable entirely?

Reply:
I assume you mean the "score_match"?

Reply:
Erm, ok, never mind he second part -- I assumed you were matching the entire text; as you matching individual lines, that would not work. Just rename `expected_results` to something more representative of it's actual function.